features/vaadin-surveillance-views/src/main/java/org/opennms/features/vaadin/surveillanceviews/ui/SurveillanceView.java

```diff
@@ -0,0 +1,360 @@
+                        setView(name);
+                    }
+                }
+            });
+
+            /**
+             * add the components...
+             */
+            addComponent(m_label);
+            addComponent(m_nativeSelect);
+
+            /**
+             * ...and align the selection box to the right
+             */
+            setComponentAlignment(m_nativeSelect, Alignment.MIDDLE_RIGHT);
+        }
+
+        /**
+         * Sets the caption text.
+         *
+         * @param text the text to be used
+         */
+        public void setCaptionText(String text) {
+            m_label.setCaption(text);
+        }
+
+        /**
+         * Selects a vie wwith the given name.
+         *
+         * @param name the name of the view to be selected
+         */
+        public void select(String name) {
+            m_nativeSelect.select(name);
+        }
+
+        /**
+         * Returns the select box instance used by this component.
+         *
+         * @return the select box instance
+         */
+        public NativeSelect getNativeSelect() {
+            return m_nativeSelect;
+        }
+    }
+
+    /**
+     * the surveillance view service used
+     */
+    private SurveillanceViewService m_surveillanceViewService;
+    /**
+     * the surveillance view table instance
+     */
+    private SurveillanceViewTable m_surveillanceViewTable;
+    /**
+     * the view to be displayed
+     */
+    private View m_view;
+    /**
+     * the table header instance
+     */
+    private SurveillanceViewTableHeader m_surveillanceViewTableHeader;
+    /**
+     * layouts for the surveillance view itself and the dependent detail tables
+     */
+    private VerticalLayout upperLayout, lowerLayout;
+    /**
+     * flag whether this component should display a dashboard
+     */
+    private boolean m_dashboard = false;
+    /**
+     * flag whether links should be enabled
+     */
+    private boolean m_enabled = true;
+    /**
+     * countdown for the refresh timer
+     */
+    private int m_countdown;
+    /**
+     * flag whether refreshing is enabled
+     */
+    private boolean m_refreshEnabled = false;
+
+    /**
+     * Constructor for creating new instances of this component.
+     *
+     * @param selectedView            the view to be used
+     * @param surveillanceViewService the surveillance view service
+     * @param dashboard               should the dashboar dbe displayed?
+     * @param enabled                 should links be enabled?
+     */
+    public SurveillanceView(View selectedView, SurveillanceViewService surveillanceViewService, boolean dashboard, boolean enabled) {
+        /**
+         * set the fields
+         */
+        this.m_surveillanceViewService = surveillanceViewService;
+        this.m_view = selectedView;
+        this.m_surveillanceViewTableHeader = new SurveillanceViewTableHeader();
+        this.m_dashboard = dashboard;
```

Comment:
The surveillance view encapsulates the SurveillanceViewTable and if needed also the dependent detail tables like outages, notifications and graphs.